src/main/java/org/jabref/model/openoffice/style/CitationLookupResult.java

```diff
@@ -0,0 +1,48 @@
+package org.jabref.model.openoffice.style;
+
+import java.util.Objects;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+
+public class CitationLookupResult {
+
+    public final BibEntry entry;
+    public final BibDatabase database;
+
+    public CitationLookupResult(BibEntry entry, BibDatabase database) {
+        Objects.requireNonNull(entry);
+        Objects.requireNonNull(database);
+        this.entry = entry;
+        this.database = database;
+    }
+
+    /**
+     * Note: BibEntry overrides Object.equals, but BibDatabase does not.
+     *
+     *       Consequently, {@code this.database.equals(that.database)} below
+     *       is equivalent to {@code this.database == that.database}.
+     *
+     *       Since within each GUI call we use a fixed list of databases, it is OK.
+     */
+    @Override
+    public boolean equals(Object otherObject) {
+        if (otherObject == this) {
+            return true;
+        }
+        if (!(otherObject instanceof CitationLookupResult)) {
+            return false;
+        }
+        CitationLookupResult that = (CitationLookupResult) otherObject;
+        return this.entry.equals(that.entry) && this.database.equals(that.database);
```

Comment:
Object.equals(..)

Reply:
changed to 
```java
return Objects.equals(this.entry, that.entry) && Objects.equals(this.database, that.database);
```
Is this what you meant to suggest?